flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMasterServiceLeadershipRunner.java

```diff
@@ -489,19 +522,29 @@ private boolean isRunning() {
     }
 
     private void runIfValidLeader(
```

Comment:
In most of the cases, we're just creating the log message as a fallback. We could reduce redundant code by adding the following helper method:
```
    private void runIfValidLeader(
            UUID expectedLeaderId, Runnable action, String noLeaderFallbackCommandDescription) {
        runIfValidLeader(
                expectedLeaderId,
                action,
                () ->
                        printLogIfNotValidLeader(
                                noLeaderFallbackCommandDescription, expectedLeaderId));
    }
```

Reply:
Fixed.